nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -226,8 +236,21 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
   // Transform path to robot base frame
   auto transformed_plan = transformGlobalPlan(pose);
 
-  // Find look ahead distance and point on path and publish
-  const double lookahead_dist = getLookAheadDistance(speed);
+  double lookahead_dist = getLookAheadDistance(speed);  
+
+  // Check for reverse driving
+  if(reverse_driving_)
+  {
+    // Cusp check
+    double orientation_change_dist_= checkOrientation(pose);
+
+    // Calculate the carrot pose based on the lookahead distance and/or the orientation_change_dist
+    if(orientation_change_dist_ < lookahead_dist)
```

Comment:
I do not understand what this is accomplishing

Reply:
This is the part where I compare the lookahead distance with distance to the cusp. Till the point where the lookahead distance is smaller than the cusp, I continue to use the lookahead distance and once when the distance to the cusp is less than the lookahead distance, then I just use the distance to the cusp as the lookahead distance for holding the carrot to the cusp. By this way, we do not have to unnecessarily have long lookahead distances.

Reply:
Got it - makes sense, that comment really doesn't make much sense though. Please update that comment to explain that better / more concisely. `if the lookahead distance is further than the cusp, use the cusp distance instead`